File: patch_preload.py
import re
with open('electron/preload.cjs', 'r') as f:
    code = f.read()

if "forceFocus:" not in code:
    code = code.replace("closeApp: () => ipcRenderer.send('close-app'),", "closeApp: () => ipcRenderer.send('close-app'),\n  forceFocus: () => ipcRenderer.send('force-focus'),")
    with open('electron/preload.cjs', 'w') as f:
        f.write(code)
    print("Patched preload.cjs")
else:
    print("Already patched")

File: patch_preload.js
const fs = require('fs');
let code = fs.readFileSync('electron/preload.cjs', 'utf-8');
code = code.replace(
  "onTriggerVoiceInput: (callback) => ipcRenderer.on('trigger-voice-input', () => callback())",
  "onTriggerVoiceInput: (callback) => { ipcRenderer.removeAllListeners('trigger-voice-input'); ipcRenderer.on('trigger-voice-input', () => callback()); }"
);
fs.writeFileSync('electron/preload.cjs', code);
